### PR TITLE
Input height consistency

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [ ] Addresses an existing issue: #1394

image

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Looks like I've got some work to do on the PeoplePicker and persona size to make all of this fit into a 32px high picker.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx [hunk 1]
```diff
@@ -56,7 +56,11 @@ export class SearchBox extends BaseComponent<ISearchBoxProps, ISearchBoxState> {
         }) }
         { ...{ onFocusCapture: this._onFocusCapture } }
       >
-        <i className={ css('ms-SearchBox-icon', 'ms-Icon', 'ms-Icon--Search', styles.icon) }></i>
+        <div
+          className={ css('ms-SearchBox-icon', styles.icon) }
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx]: wrapped icon in this div to be consistent with right side icon

File: packages/office-ui-fabric-react/src/components/TextField/TextField.scss [hunk 2]
```diff
@@ -71,7 +71,7 @@
   font-size: $ms-font-size-m;
   color: $ms-color-neutralPrimary;
   height: 32px;
-  @include padding(0, 12px, 0, 12px);
+  @include padding(0, 12px, 1px, 12px);
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.scss]: 1px bottom padding seems to be our consistent text field v alignment. I.E. line up dropdown, text field, search and picker and this 1px makes all 4 align.

**Comment by micahgodbolt** [comment]
Got a thumbs up from @leddie24 that his work with the people picker won't be affected by this change. So after review we should be good to merge.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 3]
```diff
@@ -7,17 +7,62 @@
 // --------------------------------------------------
 // Search box styles
 
-
-$SearchBox-field-padding-left: 10px;
-$SearchBox-height: 36px;
-
 .root {
   @include ms-font-m;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: no change file

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 4]
```diff
@@ -7,17 +7,62 @@
...
   @include ms-u-normalize;
   color: $ms-color-neutralPrimary;
   position: relative;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  align-items: stretch;
   margin-bottom: 10px;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: is this margin-bottom supposed to be there? Are they all consistent? I really would not expect it.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 5]
```diff
@@ -7,17 +7,62 @@
...
+.icon {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  flex-shrink: 0;
+  color: $ms-color-neutralSecondaryAlt;
+  font-size: 16px;
+  flex-basis: 32px;
+  text-align: right;
+  transition: flex-basis $ms-duration1;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: why is this transition here?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 6]
```diff
@@ -7,17 +7,62 @@
...
+
+.icon {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  flex-shrink: 0;
+  color: $ms-color-neutralSecondaryAlt;
+  font-size: 16px;
+  flex-basis: 32px;
+  text-align: right;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: This isn't RTL friendly.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 7]
```diff
@@ -7,17 +7,62 @@
...
+  transition: flex-basis $ms-duration1;
+  color: $ms-color-themePrimary
+}
+
+// Input Field
+.field {
+  @include ms-u-normalize;
+  @include padding(0px, 0, 1px, 8px);
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: hmmm. this 1px bottom seems gross... is it really needed? why?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 8]
```diff
@@ -7,17 +7,62 @@
...
+  font-weight: inherit;
+  font-family: inherit;
+  font-size: inherit;
+  color: $ms-color-black;
+  background-color: transparent;
+  flex-grow: 1;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  &::-ms-clear {
+    display: none;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: i have no idea what this is for...

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 4 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: that was already there. I'd certainly love to get rid of it, but out of scope of this update.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 5 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: to match original transition of input box sliding to fit where icon was

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 9]
```diff
@@ -38,12 +83,12 @@ $SearchBox-height: 36px;
...
-  .field {
-    @include padding-left(8px);
-  }
-
   .icon {
-    display: none;
+    flex-basis: 4px;
+    transition: flex-basis $ms-duration1;
+     & > * {
+      display: none;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: uh this seems bad. globally hiding all immediate children? what for?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx [hunk 10]
```diff
@@ -56,7 +56,11 @@ export class SearchBox extends BaseComponent<ISearchBoxProps, ISearchBoxState> {
...
+        >
+          <i className={ css('ms-Icon', 'ms-Icon--Search') }></i>
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx]: Maybe use an Icon component?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 6 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: True true. It's a bit messy given the redlines I need to match. I'll see what I can do that'll flip with flex and not need direction.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 7 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: yeah.. i need to take a look at all of the inputs and how they achieve their baselines. This 1px was needed to make it match with other designs. A pass over all inputs might be able to find a way to create them all consistently and avoid yucky hacks like that.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 8 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: This was added in a previous commit. Edge adds a clear button to inputs by default. But it's not keyboard accessible. So we chose to hide it and make functionality consistent between browsers.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 9 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: the search icon is hidden when the search is active. by setting display to none I can then reduce the flex basis of the parent and animate the change.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx (hunk 10 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx]: not a bad idea. I'll hack on this at tomorrow's bug bash and see what  I can do

**Comment by micahgodbolt** [comment]
Here's the redlines for the searchfield i'm trying to match 

microsoftteams-image 5

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 6 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: Fixed this. will push shortly

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss (hunk 9 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: back to this...i did it for a couple reasons. I can't simply display none the icon container as that'd break the animation. But I need to hide the contents of the icon container. Right now that's an expected class name, but if we ever provide a method for overriding the icon, that class might change. 

The only other option would be to wrap the rendered icon in yet another div, and target that. This seemed simpler.

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 11]
```diff
@@ -7,17 +7,68 @@
...
 
-
-$SearchBox-field-padding-left: 10px;
-$SearchBox-height: 36px;
-
 .root {
   @include ms-font-m;
   @include ms-u-normalize;
   color: $ms-color-neutralPrimary;
   position: relative;
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: do you still need this position:relative?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 12]
```diff
@@ -7,17 +7,67 @@
...
+  flex-direction: column;
+  justify-content: center;
+  flex-shrink: 0;
+  color: $ms-color-neutralSecondaryAlt;
+  font-size: 16px;
+  width: 32px;
+  text-align: center;
+  transition: width $ms-duration1;
+  color: $ms-color-themePrimary;
+  & > * {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: avoid wildcards

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss [hunk 13]
```diff
@@ -7,17 +7,67 @@
...
+  font-family: inherit;
+  font-size: inherit;
+  color: $ms-color-black;
+  background-color: transparent;
+  flex-grow: 1;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  // This padding forces the text placement to round up.
+  padding-bottom: .5px;
+  &::-ms-clear {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss]: add comment that this is to hide the clear button in IE11

**Pull request merged**